helsingborg-stad/helsingborg-io-sls-api
# feat(get-attachment): add lambda for downloading an attachment

## What this does:

Adds a lambda for downloading a users attachments, via a get request against /users/me/attachments/{filename}. The response is directly the binary file (which was a little tricky to get working correctly), which should make it easy to consume as you don't have to do any special decoding. 

This is probably useful functionality in general, but right now it's for the ability to resume stickprov.

## Implementation details

The lambda itself is rather straightforward, we just get the file from the S3 bucket, and return a response where the body is the file data in base64 encoded form, together with the minimally required headers and other properties needed. Then two serverless-plugins are used to handle the AWS setup required for binary files. To this end, some configuration is added to the serverless.common file. It's the second one of these plugins that makes the returned response into a binary file. 

## How to test

You'll need to run yarn in the users-api folder, to install the sls plugins. Then deploy the users-api (or just the getAttachment lambda).

Then, check that you have some file (of type jpg, png or pdf) in your attachment S3 bucket, for example let's say you have a file under "201111111111/test.png". Then, with an Authorization token with pnr  201111111111, a get request against "/users/me/attachments/test.png" with the additional header "Accept: image/png" should return the binary png file correctly, and it should show up as an image in Postman/Insomnia. 

(I tried getting rid of the need to send the Accept header, but that seems difficult.)
